ee/server/service/teams.go

```diff
@@ -459,9 +467,15 @@ func (svc Service) createTeamFromSpec(ctx context.Context, spec *fleet.TeamSpec,
 func (svc Service) editTeamFromSpec(ctx context.Context, team *fleet.Team, spec *fleet.TeamSpec, secrets []*fleet.EnrollSecret) error {
 	team.Name = spec.Name
 	team.Config.AgentOptions = spec.AgentOptions
-	if spec.Features != nil {
-		team.Config.Features = *spec.Features
+
+	// replace (don't merge) the features with the new ones, using a config
+	// that has the global defaults applied.
+	features, err := unmarshalWithGlobalDefaults(spec.Features)
```

Comment:
check for `spec.Features != nil`?

Reply:
yeah, that one is a bit tricky, because if it's `nil` you want a `fleet.Features{}.ApplyDefaultsForNewInstalls()`. This is different from the team creation, where if it's `nil` you want to copy the global config values.